Attempting to delete a non-existent folder returns 500

**What happened**:
Using the [Delete Folder](https://grafana.com/docs/grafana/latest/http_api/folder/#delete-folder) API endpoint to attempt to remove a non-existent folder returns a `500` error.

Request: `DELETE /api/folders/something`
Response: `{"message":"Folder API error"}`

**What you expected to happen**:
This endpoint should return `404` if the folder doesn't exist.

**How to reproduce it (as minimally and precisely as possible)**:
```
curl -X DELETE -H "Authorization: Bearer [REDACTED]" https://grafana.example.com/api/folders/something
```

**Anything else we need to know?**:
Logs show:

```
lvl=eror msg="Folder API error" logger=context userId=0 orgId=1 uname= error="found 0 folders, while expecting at most one"
```

The message seems to originate [here](https://github.com/grafana/grafana/blob/v8.2.5/pkg/services/libraryelements/database.go#L694).

**Environment**:
- Grafana version: 8.2.5

i am able to reproduce the issue with error code 500.
```
Successful POST request | AssertionError: expected response to have status code 200 but got 500
```

--- a/pkg/services/libraryelements/database.go
+++ b/pkg/services/libraryelements/database.go
@@ -692,9 +692,15 @@ func (l *LibraryElementService) deleteLibraryElementsInFolderUID(c context.Conte
 		if err != nil {
 			return err
 		}
+
+		if len(folderUIDs) == 0 {
+			return models.ErrFolderNotFound
+		}
+
 		if len(folderUIDs) != 1 {
 			return fmt.Errorf("found %d folders, while expecting at most one", len(folderUIDs))
 		}
+
 		folderID := folderUIDs[0].ID
 
 		if err := l.requirePermissionsOnFolder(c, signedInUser, folderID); err != nil {

--- a/pkg/services/libraryelements/libraryelements_test.go
+++ b/pkg/services/libraryelements/libraryelements_test.go
@@ -67,6 +67,12 @@ func TestDeleteLibraryPanelsInFolder(t *testing.T) {
 			require.EqualError(t, err, ErrFolderHasConnectedLibraryElements.Error())
 		})
 
+	scenarioWithPanel(t, "When an admin tries to delete a folder uid that doesn't exist, it should fail",
+		func(t *testing.T, sc scenarioContext) {
+			err := sc.service.DeleteLibraryElementsInFolder(sc.reqContext.Req.Context(), sc.reqContext.SignedInUser, sc.folder.Uid+"xxxx")
+			require.EqualError(t, err, models.ErrFolderNotFound.Error())
+		})
+
 	scenarioWithPanel(t, "When an admin tries to delete a folder that contains disconnected elements, it should delete all disconnected elements too",
 		func(t *testing.T, sc scenarioContext) {
 			command := getCreateVariableCommand(sc.folder.Id, "query0")

